### PR TITLE
task 9361: asset wallet: handle event

### PR DESCRIPTION
**Pull request by vanpd-0859**
task 9361: asset wallet: handle event

### Issue Link :link:

https://edu-redmine.sun-asterisk.vn/issues/9361
### Goals :soccer:

- Click Send button to go to Send Transaction Screen
- Click Receive button to go to Receive Transaction Screen
- Click copy image button to copy asset address to clipboard
- Click an transaction cell to go to Transaction Detail Screen
- Click Graph nav image button to go to Asset Detail Screen
### Implementation Details :construction:

ezgif-5-ee7cfd8132fe

### CODE REVIEW

**Comment by vanpd-0859** [comment]
ready

File: CryptoWallet/Data/Source/Remote/API/Response/CheckCronJobResponse.swift [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+//  CryptoWallet
+//
+//  Created by Phan Dinh Van on 4/14/19.
+//  Copyright © 2019 Phan Dinh Van. All rights reserved.
+//
+
+import Foundation
+import ObjectMapper
+
+final class CheckCronJobResponse: Mappable {
```
Comment (dongsonbk58) [inline CryptoWallet/Data/Source/Remote/API/Response/CheckCronJobResponse.swift]: nên chuyển thành CronJobResponse thì sẽ ok hơn là thêm check vô

File: CryptoWallet/Scenes/Home/Wallet/Asset/AssetHeader.swift [hunk 2]
```diff
@@ -104,12 +105,35 @@ final class AssetHeader: UIView {
     }
     
     @IBAction private func handleSendButtonTapped(_ sender: Any) {
+        guard let viewController = viewController else {
+            return
+        }
+        let sendTransactionController = SendTransactionViewController.instantiate()
+        viewController.navigationItem.backBarButtonItem = UIBarButtonItem(title: "Back",
```
Comment (dongsonbk58) [inline CryptoWallet/Scenes/Home/Wallet/Asset/AssetHeader.swift]: sao không chuyển vô set trong vc, check cả mấy cái bên dứoi nhé

File: CryptoWallet/Scenes/Login/RecoverWallet/RecoverWalletViewController.swift [hunk 3]
```diff
@@ -55,11 +55,24 @@ final class RecoverWalletViewController: UIViewController {
...
-                cronJobRepository.trackWallet(address: "0x1EF5B7C7d91adA52a5F4E32085F4adFAD5Ec3F6a") { (result) in
+                cronJobRepository.checkWallet(address: wallet.walletAddress) { (result) in
                     switch result {
-                    case .success(let cronJobResponse):
-                        if let cronJobResponse = cronJobResponse {
-                            print(cronJobResponse.message)
+                    case .success(let checkCronJobResponse):
+                        if let checkCronJobResponse = checkCronJobResponse {
+                            if checkCronJobResponse.description.contains(CronJobAPI.notExistEvent) {
+                                self.cronJobRepository.trackWallet(address: wallet.walletAddress) { (result) in
```
Comment (dongsonbk58) [inline CryptoWallet/Scenes/Login/RecoverWallet/RecoverWalletViewController.swift]: if let checkCronJobResponse = checkCronJobResponse, checkCronJobResponse.description.contains(CronJobAPI.notExistEvent) {}

**Comment by vanpd-0859** [comment]
ready

**Review by dongsonbk58 — APPROVED** [review]

File: CryptoWallet/Scenes/Home/Wallet/Asset/AssetHeader.swift [hunk 4]
```diff
@@ -104,12 +105,27 @@ final class AssetHeader: UIView {
     }
     
     @IBAction private func handleSendButtonTapped(_ sender: Any) {
+        guard let viewController = viewController else {
+            return
+        }
+        let sendTransactionController = SendTransactionViewController.instantiate()
```
Comment (tienpx3010) [inline CryptoWallet/Scenes/Home/Wallet/Asset/AssetHeader.swift]: Sử dụng closure

**Comment by vanpd-0859** [comment]
ready

File: CryptoWallet/Scenes/Home/Wallet/Asset/AssetHeader.swift (hunk 4 above)
Comment (tienpx3010) [reply CryptoWallet/Scenes/Home/Wallet/Asset/AssetHeader.swift]: var sendButtonTapped: (() -> Void)?
sendButtonTapped?()
vc.sendButtonTapped = {

}

File: CryptoWallet/Scenes/Home/Wallet/Asset/AssetHeader.swift [hunk 5]
```diff
@@ -104,12 +107,21 @@ final class AssetHeader: UIView {
...
+        let receiveTransactionController = ReceiveTransactionViewController.instantiate()
+        pushViewController?(receiveTransactionController, true)
     }
     
     @IBAction private func handleCopyAddressTapped(_ sender: Any) {
+        guard let address = Wallet.sharedWallet?.walletAddress else {
+            return
+        }
+        UIPasteboard.general.string = address
+        makeToast?("Address copied")
```
Comment (tienpx3010) [inline CryptoWallet/Scenes/Home/Wallet/Asset/AssetHeader.swift]: Tạo Constants trong class AssetHeader
enum Constants {
         static let copySuccessText = "Address copied"
}

**Review by tienpx3010 — APPROVED** [review]

**Pull request merged**